bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/cache/TTSResult.java

```diff
@@ -0,0 +1,521 @@
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.StandardOpenOption;
+import java.util.EnumSet;
+import java.util.Properties;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.audio.AudioFormat;
+import org.openhab.core.audio.AudioStream;
+import org.openhab.core.audio.FixedLengthAudioStream;
+import org.openhab.core.voice.TTSException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A cached sound resulting from a call to a TTSService
+ * All objects from this class form a linked list,
+ * next and previous value allow navigation through it.
+ * This class adds also the capability to serve multiple sinks concurrently
+ * without asking already retrieved data to the wrapped stream.
+ *
+ * @author Gwendal Roulleau - Initial contribution
+ */
+@NonNullByDefault
+public class TTSResult {
+
+    private final Logger logger = LoggerFactory.getLogger(TTSResult.class);
+
+    private static final String PROPERTY_TEXT = "text";
+    private static final String PROPERTY_FORMAT_BIGENDIAN = "bigEndian";
+    private static final String PROPERTY_FORMAT_BITDEPTH = "bitDepth";
+    private static final String PROPERTY_FORMAT_BITRATE = "bitRate";
+    private static final String PROPERTY_FORMAT_FREQUENCY = "frequency";
+    private static final String PROPERTY_FORMAT_CHANNELS = "channels";
+    private static final String PROPERTY_FORMAT_CODEC = "codec";
+    private static final String PROPERTY_FORMAT_CONTAINER = "container";
+
+    /**
+     * Arbitrary chunk size. Small is less latency but more small calls and CPU load.
+     */
+    private static final int CHUNK_SIZE = 10000;
+
+    /**
+     * Take count of the number of {@link AudioStreamCacheWrapper} currently using this {@link TTSResult}
+     */
+    private Integer countAudioStreamClient = 0;
+
+    /**
+     * A unique key to identify the result
+     * (used to build the filename)
+     */
+    private final String key;
+
+    private String text;
+
+    // The inner AudioStream, with a supplier
+    @Nullable
+    private AudioStreamSupplier ttsAudioStreamSupplier;
+    @Nullable
+    private AudioStream ttsAudioStream;
+    @Nullable
+    private AudioFormat audioFormat;
+
+    // The cache files where the sound and its info are stored :
+    private File soundFile;
+    private File infoFile;
+    private long currentSize = 0;
+    private boolean completed;
+
+    @NonNullByDefault({}) // file channel could not be null when we read or write from the wrapper
+    private FileChannel fileChannel;
+    private Lock fileOperationLock = new ReentrantLock();
+
+    /**
+     * These references link {@link TTSResult} between each other, allowing LRU cache to work.
+     */
+    @Nullable
+    private TTSResult next = null;
```

Comment:
this is probably not needed in a `LinkedHashMap`

Reply:
Yes, done.